backend/tests/integration-async/editor/test_editor_api.py

```diff
@@ -76,6 +95,45 @@ def test_get_revisions(inspire_app, clean_celery_session, record_with_two_revisi
     assert result[1]["user_email"] == "system"
 
 
+def test_get_revisions_hidden_collection_user(
+    inspire_app, clean_celery_session, hidden_record_with_two_revisions
+):
+    user = create_user(role="user")
+    with inspire_app.test_client() as client:
+        login_user_via_session(client, email=user.email)
+        response = client.get(
+            "/api/editor/literature/111/revisions", content_type="application/json"
+        )
+        assert response.status_code == 403
+        logout(client)
+
+
+def test_get_revisions_hidden_collection_user_read(
+    inspire_app, clean_celery_session, hidden_record_with_two_revisions
+):
+    user_read = create_user(role="hep-hidden-read")
+    with inspire_app.test_client() as client:
+        login_user_via_session(client, email=user_read.email)
+        response = client.get(
+            "/api/editor/literature/111/revisions", content_type="application/json"
+        )
+        assert response.status_code == 200
+        logout(client)
+
+
+def test_get_revisions_hidden_collection_user_write(
+    inspire_app, clean_celery_session, hidden_record_with_two_revisions
+):
+    user_readwrite = create_user(role="hep-hidden-read-write")
+    with inspire_app.test_client() as client:
+        login_user_via_session(client, email=user_readwrite.email)
+        response = client.get(
+            "/api/editor/literature/111/revisions", content_type="application/json"
+        )
+        assert response.status_code == 200
+        logout(client)
+
```

Comment:
I would add a test for curators as well, just to make sure that this case works as well